### ISSUE TITLE
Umbrella Ticket: Have working main page with statistics and hopefully working login.

### ISSUE THREAD

**Issue by 0404btruong** [0404btruong/p2-hardstuck#7]
- Brandon: https://github.com/0404btruong/p2-hardstuck/issues/7#issuecomment-853129828
- David: https://github.com/0404btruong/p2-hardstuck/issues/7#issuecomment-853197569
- Gavin:  https://github.com/0404btruong/p2-hardstuck/issues/7#issuecomment-853102100
- Kian: https://github.com/0404btruong/p2-hardstuck/issues/7#issuecomment-852669069
- Cody: https://github.com/0404btruong/p2-hardstuck/issues/7#issuecomment-853196294

**Comment by ghost** [comment]
Kian: 


**Deployed**
- Login/Signup/Logout functions [code link](https://github.com/0404btruong/p2-hardstuck/blob/a885ecee3c5052244ff76a533ae155a04e868c2f/main.py#L31)
[code link 2](https://github.com/0404btruong/p2-hardstuck/blob/a885ecee3c5052244ff76a533ae155a04e868c2f/main.py#L79)

**In progress**
- fully working API integration to display in the table (top songs from each music streaming platform) which will require a database to store the API data for display


Comments:
Kian (Self Grade): Clear criteria for in progress ticket and what is needed to achieve it and complete project, deployed shows specific code (tangibles) that connect to work previously completed

Brandon: Good tangibles and explanation of tangibles, maybe just explain what you plan to do to finish the in progress. Will you be making the database yourself or working with others?

Cody: Good updates on work status, examples for in progress would be good

David: Current work looks good, tangibles easy to follow, plans for future line up

Gavin: Tangibles are a little hard to follow. Maybe link it something a little more specific and explain what it is you did?? your first link was good and showed good explanation of work you have been doing. 2nd link took me to my email subscription. not sure why

**Comment by Uhpachee** [comment]
> Kian:
> 
> **Deployed**
> 
> * Login/Signup/Logout functions [code link](https://github.com/0404btruong/p2-hardstuck/blob/a885ecee3c5052244ff76a533ae155a04e868c2f/main.py#L31)
>   [code link 2](https://github.com/0404btruong/p2-hardstuck/blob/a885ecee3c5052244ff76a533ae155a04e868c2f/main.py#L79)
> 
> **In progress**
> 
> * fully working API integration to display in the table (top songs from each music streaming platform)

**Comment by gavintheriault1** [comment]
Gavin:

add CSS to login, subscription, and sign up pages

Deployed: Login page

Tangible: https://github.com/0404btruong/p2-hardstuck/blob/main/templates/login.html

In progress: 

Almost wrapped up with custom css for subscription and the sign up pages. May add a gradient or changing color scheme. 

Comments: Self grade: Shows tangible and evidence of work done and shows what I have in progress at the moment. Clear and concise. 

Kian: Clear tangibles and in progress is detailed, maybe describe the deployed a bit more (front end or back end? etc.)

Brandon:  Good work on the login, good tangibles, like Kian said, maybe just explain what you did specifically, like what work in the CSS

Gavin: I think your tangibles are a little hard to follow. Just links us to main.py. maybe highlight the pieces you did?? 

David: Formatting is good, clarify tangibles, link to location on deployed site?

Cody: As David said, a link to the login page would be nice

**Comment by 0404btruong** [comment]
Brandon:

**Deployed**: Website front page, subscription section button, table, css
Tangible: https://github.com/0404btruong/p2-hardstuck/blob/main/templates/index.html
In progress: working on linking all other pages and working with backend people to make table more interactive

Self grade: i have tangibles and I have a plan for the future. The main thing is just doing it at this point, but i got to the point. 

Comments:
Kian: Detailed list of deployed, maybe put more specifics in the in progress ticket (which pages are you linking? etc.)

Gavin: I think make the table more interactive would be interesting. How do you plan on doing that? I like how your tangibles are easy to follow and show clear organization of the work you have done 

Cody: Very clear and concise description of tangible

David: Looks good, should integrate nicely with backend that I am working on, link to location on deployed site?

**Comment by Cody-Peng** [comment]
Cody:

Deployed: Updates to the website
(https://github.com/0404btruong/p2-hardstuck/blob/06b99e5a48a4a785bfc082a7f5f85af5373137cf/people/Cody/templates/easteregg.html)
(http://18.220.18.125/people/Cody/easteregg)

In progress: Easter egg for website, along with correct redirect links (making them work)

Self grade: Lack of tangibles but will improve by next week (note: need to correctly reroute) 

Comments:
Kian: No tangibles, very vague deployed, lots in progress and specific assignments

Gavin: Make sure to get some stuff into the deployed section and in review. Want to see the tangibles and work you've done.

David: Will look forward to what will come in the future

Brandon: good deployed work, will look forward to rerouted stuff and updated easter egg

**Comment by davidk-im** [comment]
David
**Deployed**: [Code](https://github.com/0404btruong/p2-hardstuck)
**Tangibles**: [Code](https://github.com/0404btruong/p2-hardstuck/blob/b414777a51cf80983c06665530bcc022993db6a1/people/David/Song-class.py#L87)
- [x] Class "Song" for objects made for songs ([Code](https://github.com/0404btruong/p2-hardstuck/blob/main/people/David/Song-class.py)) 
- [x] access pure data from api, pull data from json
- [x] use methods from classes to calculate basic data from pure data from api

**In Progress**:
- [ ] link everything to frontend
- [ ] make sure that data is transferred correctly
- [ ] Class "people" for users

**Self grade**: tangibles link to code, made progress in meaningful project contribution

Comments:
Kian: Very organized, very clear and specific criteria for both deployed and in progress, tangibles present, very detailed

Brandon: Very good tangibles and deployed section, I will make sure to work with you to link to frontend to get this done.

Cody: Very neat formatting and easy to see progression

Gavin: Well David... you never cease to impress me. Your work with the api is great. Can't wait to see when you link it all to the front end
